---
title: "May15-2018_Results"
author: "Sabah Ul-Hasan"
output:
  pdf_document: default
  html_document: default
---

https://github.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal

Figure                   
1     line  20                         
2     line  96          
3     line  201               
4i    line  558                   
4ii   line  849                   
5     line  929                   
Supp  line  1160      

Colors      https://www.rapidtables.com/web/color/RGB_Color.html , colorbrewer2.org

Figure 1: Sampling site locations
```{r}
######### Packages ###########
# Upload the respective packages
devtools::install_github("hadley/ggplot2") # Need for main figure
devtools::install_github("dkahle/ggmap") # Need for main figure
install.packages("maps")
install.packages("mapproj")
install.packages("mapdata")
install.packages("maptools")
install.packages("sp")
install.packages("ggsn")
# Libraries
library("ggplot2")
library("ggmap")
library("maps")
library("mapproj")
library("mapdata")
library("maptools")
library("sp")
library("ggsn")
# Package versions
packageVersion("ggplot2") # v2.2.1
packageVersion("ggmap") # v2.7.900
packageVersion("maps") # v3.2.0
packageVersion("mapproj") # v1.2.5
packageVersion("mapdata") # v2.2.6
packageVersion("maptools") # v0.9.2
packageVersion("sp") # v1.2.5
packageVersion("ggsn") # v0.4.0
######### Packages ###########



######### Data and Clean-up ###########
# Upload site metadata .csv 
install.packages("RCurl")
library("RCurl")
packageVersion("RCurl") # v1.95.4.10  *allows you to download from GitHub
Jun7_Sites=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/Jun7-2016_CollectionMetadata.csv"))
# Subset for lat and lon only
Sites_LatLon=Jun7_Sites[ , 1:3] # [ rows, columns ]
######### Data and Clean-up ###########



######### Create Maps for Figure 1 ###########
# Identify lon and lat ranges, create buffer around
xlim=range(Sites_LatLon$Lon)+c(-.01, 0.01)
# -116.9684 -116.9242
ylim=range(Sites_LatLon$Lat)+c(-0.01, 0.01)
# 32.22645 32.26754

# Create broad-scale map (inset)
Broad=get_map(location=c(-118.5,27,-113,33), zoom=7, maptype="terrain")
# satellite shows bathymetry
Broad_Map=ggmap(Broad) 
ggsave("Figure1_InsetMap.png", dpi = 300)

# Create zoomed-in map (main figure)
Zoom=get_map(location=c(-116.9685,32.22644,-116.9241,32.26755), maptype="roadmap")
Zoom_Map=ggmap(Zoom) 
Zoom_Map2 = Zoom_Map + 
  geom_point(data=Sites_LatLon, aes(x=Lon, y=Lat, fill=Site_Name, shape=Site_Name), color="black", cex=4.5) + # points
  scale_fill_hue(labels=c("Minor Outlet", "Sheltered", "Major Outlet"), name=NULL) +
  scale_shape_manual(values = c(21,21,21), labels=c("Minor Outlet", "Sheltered", "Major Outlet"), name=NULL) + # define shapes
  scale_fill_manual(labels=c("Minor Outlet", "Sheltered", "Major Outlet"), values=c("#F65A00", "#68AF2B", "#0202DE")) + # MJ orange, MN green, SH blue
  theme(legend.position=0) +
  labs(x="Latitude", y="Longitude") # label axes 
Zoom_Map2 
ggsave("Figure1_Map.png", dpi = 300)
######### Create Maps for Figure 1 ###########

# Include and adjust for final Figure 1 manually
```

Figure 2: Taxa richness
```{r}
######### Packages ###########
install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # v2.2.1
install.packages("gridExtra")
library(gridExtra)
packageVersion("gridExtra") # v2.3
######### Packages ###########


######### Data and Clean-up ###########
# Caenogastropoda (cone snail) removed from all
# Contaminant Eukaryote in Prokaryote data removed from all
# Rarefied at 1000
install.packages("RCurl")
library("RCurl")
packageVersion("RCurl") # v1.95.4.10  *allows you to download from GitHub
# Prok = Prokaryotes (seawater & sediment microbes)
Prok=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_WatSed_1000.csv"))
# Euk = Eukaryotes (seawater & sediment microbes)
Euk=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_WatSed_1000.csv"))
# Metadata for Prok and Euk
Meta=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/Combined_Meta_1000.csv"))

Prok[Prok>1]=1 # Set abundance to 1 "‘>’ not meaningful for factors" still works
Euk[Euk>1]=1 # Set abundance to 1 "‘>’ not meaningful for factors" still works

Prok=merge(Meta, Prok, by = "Name") # Merge datasets
Euk=merge(Meta, Euk, by = "ID")
######### Data and Clean-up ###########



######### Statistics ###########
# Prokaryotes
Prok$richness=rowSums(Prok[,5:2204]) # Create a column with sums of row richness

aggregate(richness ~ Medium*Location, data = Prok, FUN = "mean")
Prok_aov=aov(richness ~ Medium*Location, data = Prok)

shapiro.test(residuals(Prok_aov)) # p = 0.131 (passed)
summary(Prok_aov) 
# Medium: p < 0.001, Location p = 0.900, Interaction p = 0.562
# Significant difference in medium only
# How much richer is the sediment compared to seawater? 
ProkWat = subset(Prok, Medium=="Water")
ProkWat = sum(ProkWat$richness) # 2861
ProkSed = subset(Prok, Medium=="Sediment")
ProkSed = sum(ProkSed$richness)
ProkSed = ProkSed * 400 # ( 200 g / 0.5 g = 400 ) 1624400
ProkPercent = (ProkSed/ProkWat) # Sed is 5.7 x 10^2 fold richer than seawater
# Eukaryotes
Euk$richness=rowSums(Euk[,5:1242]) # Create a column with sums of row richness

aggregate(richness ~ Medium*Location, data = Euk, FUN = "mean")
Euk_aov=aov(richness ~ Medium*Location, data = Euk)

shapiro.test(residuals(Euk_aov)) # p = 0.627 (passed)
summary(Euk_aov)
# Medium: p < 0.001, Location p = 0.110, Interaction p = 0.915
# Significant difference in medium only
# How much richer is the sediment compared to seawater? 
EukWat = subset(Euk, Medium=="Water")
EukWat = sum(EukWat$richness) # 1469
EukSed = subset(Euk, Medium=="Sediment")
EukSed = sum(EukSed$richness)
EukSed = EukSed * 400 # ( 200 g / 0.5 g = 400 ) 716000
EukPercent = (EukSed/EukWat) # Sed is 4.9 x 10^2 fold richer than seawater
######### Statistics ###########



######### Richness Plots for Figure 2 ###########
##### Figure doesn't account for for 200 mL (g) water vs 0.5 g sediment (raw)
##### Fold richness does account for this (see statistics above)
# Prokaryotes
Prok_Rich = 
  ggplot(Prok, aes(x = Location, y = richness, fill = Location, alpha = Medium)) +
  geom_boxplot(lwd=1, outlier.size = 3) +
  scale_y_continuous(name = "Bacteria and Archaea (Raw) Richness") +
  scale_fill_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  scale_alpha_manual(values=c(1,1)) +
  theme_linedraw() +
  theme(legend.position = "right", text = element_text(size=9),  axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 100)), axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0)))
Prok_Rich
ggsave("Figure2a_ProkRich.png", dpi = 300)

# Eukaryotes
Euk_Rich = 
  ggplot(Euk, aes(x = Location, y = richness, fill = Location, alpha = Medium)) +
  geom_boxplot(lwd=1, outlier.size = 3) +
  scale_y_continuous(name = "Eukaryote (Raw) Richness") +
  scale_fill_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  scale_alpha_manual(values=c(1,1)) +
  theme_linedraw() +
  theme(legend.position = "right", text = element_text(size=9),  axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 100)), axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0)))
Euk_Rich
ggsave("Figure2b_EukRich.png", dpi = 300)
######### Richness Plots for Figure 2 ###########

# Adjust for final Figure 2 manually
```

Figure 3: OTU bar graphs
```{r}
######### Packages ###########
# Upload the respective packages
install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # v2.2.1
install.packages("stringr")
library(stringr)
packageVersion("stringr") # v1.2.0
install.packages("dplyr")
library(dplyr)
packageVersion("dplyr") # v0.7.4
install.packages("tidyr")
library("tidyr")
packageVersion("tidyr") # v0.7.2
######### Packages ###########



######### Data and Clean-up ###########
# Caenogastropoda (cone snail) removed from all
# Contaminant Eukaryote in Prokaryote data removed from all
# Rarefied at 1000
install.packages("RCurl")
library("RCurl")
packageVersion("RCurl") # v1.95.4.10  *allows you to download from GitHub
# Prok = Prokaryotes (seawater & sediment microbes)
Prok=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_WatSed_1000.csv"))
# Euk = Eukaryotes (seawater & sediment microbes)
Euk=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_WatSed_1000.csv"))
# Metadata for Prok and Euk
Meta=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/Combined_Meta_1000.csv"))

Prok[Prok>1]=1 # Set abundance to 1 "‘>’ not meaningful for factors" still works
Euk[Euk>1]=1 # Set abundance to 1 "‘>’ not meaningful for factors" still works

# Merge data sets with metadata
Data_Prok=merge(Meta,Prok,by="Name")
Data_Euk=merge(Meta,Euk,by="ID")

# Subset by environment type
Sed_16S=subset(Data_Prok,Medium=="Sediment")
H2O_16S=subset(Data_Prok,Medium=="Water")
Sed_18S=subset(Data_Euk,Medium=="Sediment")
H2O_18S=subset(Data_Euk,Medium=="Water")

# Upload OTU taxa tables
# Prok Phyla and Class
ProkTaxa=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_Taxa.csv"))
# Euk Phyla and Class
EukTaxa=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_Taxa.csv"))

# Add X so all OTU match
ProkTaxa$OTU <- sub("^", "X", ProkTaxa$OTU )
EukTaxa$OTU <- sub("^", "X", EukTaxa$OTU )

# Transpose all environment type data sets
Sed_16S=as.data.frame(t(Sed_16S))
H2O_16S=as.data.frame(t(H2O_16S))
Sed_18S=as.data.frame(t(Sed_18S))
H2O_18S=as.data.frame(t(H2O_18S))

# Create a new column out of left and label as "OTU" 
Sed_16S = add_rownames(Sed_16S, "OTU")
H2O_16S = add_rownames(H2O_16S, "OTU")
Sed_18S = add_rownames(Sed_18S, "OTU")
H2O_18S = add_rownames(H2O_18S, "OTU")

Sed_16S=merge(ProkTaxa,Sed_16S,by="OTU") # 13-16 SH, 17-19 MN, 20-23 MJ
H2O_16S=merge(ProkTaxa,H2O_16S,by="OTU") # 1-4 SH, 5-8 MN, 9-12 MJ
Sed_18S=merge(EukTaxa,Sed_18S,by="OTU") # 1-3 SH, 4-6 MN, 7-9 MJ
H2O_18S=merge(EukTaxa,H2O_18S,by="OTU") # 10-13 SH, 14-17 MN, 18-21 MJ
######### Data and Clean-up ###########



######### Identify Abundant/Rare Phyla and Classes, Figure 3 ###########
########## ProkSed ##########
ProkSed1 = Sed_16S[,3:15] # remove OTU and Taxa columns
ProkSed1[,3:13] = apply(ProkSed1[,3:13], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
### Phyla ###
ProkSed2 = data.frame(apply(ProkSed1[,3:13], 2, function(x) aggregate (x~ProkSed1$Phylum, FUN=sum))) # add phylum read count per each location
ProkSed2 = ProkSed2[,c(1,2,4,6,8,10,12,14,16,18,20,22)] # drop repeat phyla
colnames(ProkSed2)[colnames(ProkSed2)=="X13.ProkSed1.Phylum"] = "Phylum"
ProkSed3 = gather(ProkSed2, "Phylum", 2:12) # aggregate by Phylum
names(ProkSed3)=c("Phylum", "Location", "Richness") # relabel columns
ProkSed_Phyla=aggregate(Richness~Phylum, data=ProkSed3, FUN=sum) # Join loci
ProkSed_Phyla # Look at table
# Top 10 Phyla (high to low) = Bacteroidetes, Proteobacteria, Planctomycetes, Acidobacteria, Actinobacteria, Chloroflexi, Verrucomicrobia, Lentisphaerae, Firmicutes, Chlorobi
# Rare (low to high, abundance 1-10) = Gracilibacteria, Hydrogenedentes, Euryarchaeota, Parcubacteria, Cloacimonetes, Atribacteria, Marinimicrobia__SAR406_clade_, PAUC34f, SHA_109, Fibrobacteres, Omnitrophica, TA06, Woesearchaeota__DHVEG_6_, Deinococcus_Thermus, Elusimicrobia, Cyanobacteria
Bacteroidetes=subset(ProkSed3, Phylum == c("Bacteroidetes"))
Proteobacteria=subset(ProkSed3, Phylum == c("Proteobacteria"))
Planctomycetes=subset(ProkSed3, Phylum == c("Planctomycetes"))
Acidobacteria=subset(ProkSed3, Phylum == c("Acidobacteria"))
Actinobacteria=subset(ProkSed3, Phylum == c("Actinobacteria"))
Chloroflexi=subset(ProkSed3, Phylum == c("Chloroflexi"))
Verrucomicrobia=subset(ProkSed3, Phylum == c("Verrucomicrobia"))
Lentisphaerae=subset(ProkSed3, Phylum == c("Lentisphaerae"))
Firmicutes=subset(ProkSed3, Phylum == c("Firmicutes"))
Chlorobi=subset(ProkSed3, Phylum == c("Chlorobi"))
# Combine data sets
ProkSed_PhylaOTU=rbind(Bacteroidetes, Proteobacteria, Planctomycetes, Acidobacteria, Actinobacteria, Chloroflexi, Verrucomicrobia, Lentisphaerae, Firmicutes, Chlorobi) 
# Create percentages so stacked bars add up to 100
ProkSed_PhylaOTU2=aggregate(Richness~Location, ProkSed_PhylaOTU, FUN=sum)
ProkSed_PhylaOTU3=merge(ProkSed_PhylaOTU, ProkSed_PhylaOTU2, by = "Location", all = T)
ProkSed_PhylaOTU3$Percentage=(ProkSed_PhylaOTU3$Richness.x/ProkSed_PhylaOTU3$Richness.y)*100
# OTU Phyla graph
ProkSed_Phylagraph = 
  ggplot(data = ProkSed_PhylaOTU3, aes(x = Location, y = Percentage, fill = Phylum)) +
  scale_fill_manual(values=c("#053061", "#2166ac", "#4393c3", "#92c5de", "#f7f7f7", "#e6f5d0", "#b8e186", "#7fbc41", "#4d9221", "#276419")) +
  geom_bar(stat = "identity")
ggsave("Figure3_ProkSed_Phyla.png", dpi = 300)
### Classes ### (use for Figure 5 heatmaps)
ProkSed4 = data.frame(apply(ProkSed1[,3:13], 2, function(x) aggregate (x~ProkSed1$Class, FUN=sum))) # add class read count per each location
ProkSed4 = ProkSed4[,c(1,2,4,6,8,10,12,14,16,18,20,22)] # drop repeat classes
colnames(ProkSed4)[colnames(ProkSed4)=="X13.ProkSed1.Class"] = "Class"
ProkSed5 = gather(ProkSed4, "Class", 2:12) # aggregate by Class
names(ProkSed5)=c("Class", "Location", "Richness") # relabel columns
ProkSed_Class=aggregate(Richness~Class, data=ProkSed5, FUN=sum) # Join loci
ProkSed_Class # Look at table
# Top 10 Class (high to low) = Sphingobacteriia, Gammaproteobacteria, Deltaproteobacteria, Planctomycetacia, Flavobacteriia, Alphaproteobacteria, Cytophagia, Acidimicrobiia, Verrucomicrobiae, OM190
# Rare (1 read) = 028H05_P_BN_P5, DEV055, JTB23, KD4_96, Latescibacteria_Incertae_Sedis_11404, LD1_PA20, LD1_PA34, Methanomicrobia, OPB54, SAR202_clade, Subgroup_26, Thermoplasmata, Unknown_Class_1002391, Unknown_Class_1002397, vadinHA49
Sphingobacteriia=subset(ProkSed5, Class == c("Sphingobacteriia"))
Gammaproteobacteria=subset(ProkSed5, Class == c("Gammaproteobacteria"))
Deltaproteobacteria=subset(ProkSed5, Class == c("Deltaproteobacteria"))
Planctomycetacia=subset(ProkSed5, Class == c("Planctomycetacia"))
Flavobacteriia=subset(ProkSed5, Class == c("Flavobacteriia"))
Alphaproteobacteria=subset(ProkSed5, Class == c("Alphaproteobacteria"))
Cytophagia=subset(ProkSed5, Class == c("Cytophagia"))
Acidimicrobiia=subset(ProkSed5, Class == c("Acidimicrobiia"))
Verrucomicrobiae=subset(ProkSed5, Class == c("Verrucomicrobiae"))
OM190=subset(ProkSed5, Class == c("OM190"))
# Combine data sets
ProkSed_ClassOTU=rbind(Sphingobacteriia, Gammaproteobacteria, Deltaproteobacteria, Planctomycetacia, Flavobacteriia, Alphaproteobacteria, Cytophagia, Acidimicrobiia, Verrucomicrobiae, OM190) 
# Create percentages so stacked bars add up to 100
ProkSed_ClassOTU2=aggregate(Richness~Location, ProkSed_ClassOTU, FUN=sum)
ProkSed_ClassOTU3=merge(ProkSed_ClassOTU, ProkSed_ClassOTU2, by = "Location", all = T)
ProkSed_ClassOTU3$Percentage=(ProkSed_ClassOTU3$Richness.x/ProkSed_ClassOTU3$Richness.y)*100
# OTU Class graph
ProkSed_Classgraph = 
  ggplot(data = ProkSed_ClassOTU3, aes(x = Location, y = Percentage, fill = Class)) +
  scale_fill_manual(values=c("#053061", "#2166ac", "#4393c3", "#92c5de", "#f7f7f7", "#e6f5d0", "#b8e186", "#7fbc41", "#4d9221", "#276419")) +
  geom_bar(stat = "identity")
ggsave("Figure3_ProkSed_Class.png", dpi = 300)
########## ProkSed ##########
########## ProkWat ##########
# ProkWat
ProkWat1 = H2O_16S[,3:16] # remove OTU and Taxa columns
ProkWat1[,3:14] = apply(ProkWat1[,3:14], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
### Phyla ###
ProkWat2 = data.frame(apply(ProkWat1[,3:14], 2, function(x) aggregate (x~ProkWat1$Phylum, FUN=sum))) # add phylum read count per each location
ProkWat2 = ProkWat2[,c(1,2,4,6,8,10,12,14,16,18,20,22,24)] # drop repeat phyla
colnames(ProkWat2)[colnames(ProkWat2)=="X1.ProkWat1.Phylum"] = "Phylum"
ProkWat3 = gather(ProkWat2, "Phylum", 2:13) # aggregate by Phylum
names(ProkWat3)=c("Phylum", "Location", "Richness") # relabel columns
ProkWat_Phyla=aggregate(Richness~Phylum, data=ProkWat3, FUN=sum) # Join loci
ProkWat_Phyla # Look at table
# Top 10 Phyla (high to low) = Proteobacteria, Bacteroidetes, Planctomycetes, Firmicutes, Verrucomicrobia, Cyanobacteria, Lentisphaerae, Actinobacteria, Spirochaetae, Woesearchaeota__DHVEG_6_
# Rare (low to high, abundance 1-10) = Armatimonadetes, Deferribacteres, Gemmatimonadetes, Latescibacteria, Microgenomates, Nitrospirae, Parcubacteria, TA06, Chloroflexi, Elusimicrobia, Omnitrophica, Chlorobi, Deinococcus_Thermus, TM6, Acidobacteria, Cloacimonetes, Fibrobacteres
Proteobacteria=subset(ProkWat3, Phylum == c("Proteobacteria"))
Bacteroidetes=subset(ProkWat3, Phylum == c("Bacteroidetes"))
Planctomycetes=subset(ProkWat3, Phylum == c("Planctomycetes"))
Firmicutes=subset(ProkWat3, Phylum == c("Firmicutes"))
Verrucomicrobia=subset(ProkWat3, Phylum == c("Verrucomicrobia"))
Cyanobacteria=subset(ProkWat3, Phylum == c("Cyanobacteria"))
Lentisphaerae=subset(ProkWat3, Phylum == c("Lentisphaerae"))
Actinobacteria=subset(ProkWat3, Phylum == c("Actinobacteria"))
Spirochaetae=subset(ProkWat3, Phylum == c("Spirochaetae"))
Woesearchaeota_DHVEG6=subset(ProkWat3, Phylum == c("Woesearchaeota__DHVEG_6_"))
# Combine data sets
ProkWat_PhylaOTU=rbind(Proteobacteria, Bacteroidetes, Planctomycetes, Firmicutes, Verrucomicrobia, Cyanobacteria, Lentisphaerae, Actinobacteria, Spirochaetae, Woesearchaeota_DHVEG6) 
# Create percentages so stacked bars add up to 100
ProkWat_PhylaOTU2=aggregate(Richness~Location, ProkWat_PhylaOTU, FUN=sum)
ProkWat_PhylaOTU3=merge(ProkWat_PhylaOTU, ProkWat_PhylaOTU2, by = "Location", all = T)
ProkWat_PhylaOTU3$Percentage=(ProkWat_PhylaOTU3$Richness.x/ProkWat_PhylaOTU3$Richness.y)*100
# OTU Phyla graph
ProkWat_Phylagraph = 
  ggplot(data = ProkWat_PhylaOTU3, aes(x = Location, y = Percentage, fill = Phylum)) +
  scale_fill_manual(values=c("#2166ac", "#4393c3", "#F65A00", "#e6f5d0", "#b8e186", "#7fbc41", "#4d9221", "#F65A00", "#276419", "#F65A00")) +
  geom_bar(stat = "identity")
ggsave("Figure3_ProkWat_Phyla.png", dpi = 300)
### Classes ### 
ProkWat4 = data.frame(apply(ProkWat1[,3:14], 2, function(x) aggregate (x~ProkWat1$Class, FUN=sum))) # add class read count per each location
ProkWat4 = ProkWat4[,c(1,2,4,6,8,10,12,14,16,18,20,22,24)] # drop repeat classes
colnames(ProkWat4)[colnames(ProkWat4)=="X1.ProkWat1.Class"] = "Class"
ProkWat5 = gather(ProkWat4, "Class", 2:13) # aggregate by Class
names(ProkWat5)=c("Class", "Location", "Richness") # relabel columns
ProkWat_Class=aggregate(Richness~Class, data=ProkWat5, FUN=sum) # Join loci
ProkWat_Class # Look at table
# Top 10 Class (high to low) = Gammaproteobacteria, Alphaproteobacteria, Flavobacteriia, Deltaproteobacteria, Bacteroidia, Sphingobacteriia, Clostridia, Planctomycetacia, Cyanobacteria, Cytophagia
# Rare (1 read) = Actinobacteria, AEGEAN_245, Anaerolineae, Ardenticatenia, Deferribacteres_Incertae_Sedis_11258, Gemmatimonadetes, Holophagae, Latescibacteria_Incertae_Sedis_11404, LD1_PA20, Lentisphaeria, Nitrospira, OPB35_soil_group, SB_5, Skagenf62, SPOTSOCT00m83, SS1_B_03_39, Subgroup_22, Unknown_Class_1002196, Unknown_Class_1002468, Unknown_Class_1002476, Unknown_Class_1002669, WCHB1_32
Gammaproteobacteria=subset(ProkWat5, Class == c("Gammaproteobacteria"))
Alphaproteobacteria=subset(ProkWat5, Class == c("Alphaproteobacteria"))
Flavobacteriia=subset(ProkWat5, Class == c("Flavobacteriia"))
Deltaproteobacteria=subset(ProkWat5, Class == c("Deltaproteobacteria"))
Bacteroidia=subset(ProkWat5, Class == c("Bacteroidia"))
Sphingobacteriia=subset(ProkWat5, Class == c("Sphingobacteriia"))
Clostridia=subset(ProkWat5, Class == c("Clostridia"))
Planctomycetacia=subset(ProkWat5, Class == c("Planctomycetacia"))
Cyanobacteria=subset(ProkWat5, Class == c("Cyanobacteria"))
Cytophagia=subset(ProkWat5, Class == c("Cytophagia"))
# Combine data sets
ProkWat_ClassOTU=rbind(Gammaproteobacteria, Alphaproteobacteria, Flavobacteriia, Deltaproteobacteria, Bacteroidia, Sphingobacteriia, Clostridia, Planctomycetacia, Cyanobacteria, Cytophagia) 
# Create percentages so stacked bars add up to 100
ProkWat_ClassOTU2=aggregate(Richness~Location, ProkWat_ClassOTU, FUN=sum)
ProkWat_ClassOTU3=merge(ProkWat_ClassOTU, ProkWat_ClassOTU2, by = "Location", all = T)
ProkWat_ClassOTU3$Percentage=(ProkWat_ClassOTU3$Richness.x/ProkWat_ClassOTU3$Richness.y)*100
# OTU Class graph
ProkWat_Classgraph = 
  ggplot(data = ProkWat_ClassOTU3, aes(x = Location, y = Percentage, fill = Class)) +
  scale_fill_manual(values=c("#2166ac", "gold", "#F65A00", "darkred", "#4393c3", "#92c5de", "#f7f7f7", "#e6f5d0", "#7fbc41", "#4d9221")) +
  geom_bar(stat = "identity")
ggsave("Figure3_ProkWat_Class.png", dpi = 300)
########## ProkWat ##########
########## EukSed ##########
# EukSed
EukSed1 = Sed_18S[,3:13] # remove OTU and Taxa columns
EukSed1[,3:11] = apply(EukSed1[,3:11], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
### Phyla ###
EukSed2 = data.frame(apply(EukSed1[,3:11], 2, function(x) aggregate (x~EukSed1$Phylum, FUN=sum))) # add phylum read count per each location
EukSed2 = EukSed2[,c(1,2,4,6,8,10,12,14,16,18)] # drop repeat phyla
colnames(EukSed2)[colnames(EukSed2)=="X1.EukSed1.Phylum"] = "Phylum"
EukSed3 = gather(EukSed2, "Phylum", 2:10) # aggregate by Phylum
names(EukSed3)=c("Phylum", "Location", "Richness") # relabel columns
EukSed_Phyla=aggregate(Richness~Phylum, data=EukSed3, FUN=sum) # Join loci
EukSed_Phyla # Look at table
# Top 10 Phyla (high to low) = Ochrophyta, Dinoflagellata, Ciliophora, Basidiomycota, Cnidaria, Florideophycidae, Annelida, Arthropoda, Apicomplexa, Unknown_Phylum_1002968
# Rare (low to high, abundance 1-10) = Tunicata, Echinodermata, Platyhelminthes, Ascomycota, Nematoda, SCM37C52, Protalveolata, SGUH942, Mollusca, Unknown_Phylum_1003810
Ochrophyta=subset(EukSed3, Phylum == c("Ochrophyta"))
Dinoflagellata=subset(EukSed3, Phylum == c("Dinoflagellata"))
Ciliophora=subset(EukSed3, Phylum == c("Ciliophora"))
Basidiomycota=subset(EukSed3, Phylum == c("Basidiomycota"))
Cnidaria=subset(EukSed3, Phylum == c("Cnidaria"))
Florideophycidae=subset(EukSed3, Phylum == c("Florideophycidae"))
Annelida=subset(EukSed3, Phylum == c("Annelida"))
Arthropoda=subset(EukSed3, Phylum == c("Arthropoda"))
Apicomplexa=subset(EukSed3, Phylum == c("Apicomplexa"))
Unknown_Phylum_1002968=subset(EukSed3, Phylum == c("Unknown_Phylum_1002968"))
# Combine data sets
EukSed_PhylaOTU=rbind(Ochrophyta, Dinoflagellata, Ciliophora, Basidiomycota, Cnidaria, Florideophycidae, Annelida, Arthropoda, Apicomplexa, Unknown_Phylum_1002968) 
# Create percentages so stacked bars add up to 100
EukSed_PhylaOTU2=aggregate(Richness~Location, EukSed_PhylaOTU, FUN=sum)
EukSed_PhylaOTU3=merge(EukSed_PhylaOTU, EukSed_PhylaOTU2, by = "Location", all = T)
EukSed_PhylaOTU3$Percentage=(EukSed_PhylaOTU3$Richness.x/EukSed_PhylaOTU3$Richness.y)*100
# OTU Phyla graph
EukSed_Phylagraph = 
  ggplot(data = EukSed_PhylaOTU3, aes(x = Location, y = Percentage, fill = Phylum)) +
  scale_fill_manual(values=c("#053061", "#2166ac", "#4393c3", "#92c5de", "#f7f7f7", "#e6f5d0", "#b8e186", "#7fbc41", "#4d9221", "#276419")) +
  geom_bar(stat = "identity")
ggsave("Figure3_EukSed_Phyla.png", dpi = 300)
### Classes ### (use for Figure 5 heatmaps)
EukSed4 = data.frame(apply(EukSed1[,3:11], 2, function(x) aggregate (x~EukSed1$Class, FUN=sum))) # add class read count per each location
EukSed4 = EukSed4[,c(1,2,4,6,8,10,12,14,16,18)] # drop repeat classes
colnames(EukSed4)[colnames(EukSed4)=="X1.EukSed1.Class"] = "Class"
EukSed5 = gather(EukSed4, "Class", 2:10) # aggregate by Class
names(EukSed5)=c("Class", "Location", "Richness") # relabel columns
EukSed_Class=aggregate(Richness~Class, data=EukSed5, FUN=sum) # Join loci
EukSed_Class # Look at table
# Top 10 Class (high to low) = Diatomea, Dinophyceae, Intramacronucleata, Agaricomycetes, Anthozoa, Polychaeta, Rhodymeniophycidae, Arachnida, Unknown_Class_1003091, Conoidasida
# Rare (abundance 1) = Ascidiacea and Polyplacophora
Diatomea=subset(EukSed5, Class == c("Diatomea"))
Dinophyceae=subset(EukSed5, Class == c("Dinophyceae"))
Intramacronucleata=subset(EukSed5, Class == c("Intramacronucleata"))
Agaricomycetes=subset(EukSed5, Class == c("Agaricomycetes"))
Anthozoa=subset(EukSed5, Class == c("Anthozoa"))
Polychaeta=subset(EukSed5, Class == c("Polychaeta"))
Rhodymeniophycidae=subset(EukSed5, Class == c("Rhodymeniophycidae"))
Arachnida=subset(EukSed5, Class == c("Arachnida"))
Unknown_Class_1003091=subset(EukSed5, Class == c("Unknown_Class_1003091"))
Conoidasida=subset(EukSed5, Class == c("Conoidasida"))
# Combine data sets
EukSed_ClassOTU=rbind(Diatomea, Dinophyceae, Intramacronucleata, Agaricomycetes, Anthozoa, Polychaeta, Rhodymeniophycidae, Arachnida, Unknown_Class_1003091, Conoidasida) 
# Create percentages so stacked bars add up to 100
EukSed_ClassOTU2=aggregate(Richness~Location, EukSed_ClassOTU, FUN=sum)
EukSed_ClassOTU3=merge(EukSed_ClassOTU, EukSed_ClassOTU2, by = "Location", all = T)
EukSed_ClassOTU3$Percentage=(EukSed_ClassOTU3$Richness.x/EukSed_ClassOTU3$Richness.y)*100
# OTU Class graph
EukSed_Classgraph = 
  ggplot(data = EukSed_ClassOTU3, aes(x = Location, y = Percentage, fill = Class)) +
  scale_fill_manual(values=c("#053061", "#2166ac", "#4393c3", "#92c5de", "#f7f7f7", "#e6f5d0", "#b8e186", "#7fbc41", "#4d9221", "#276419")) +
  geom_bar(stat = "identity")
ggsave("Figure3_EukSed_Class.png", dpi = 300)
########## EukWat ##########
# EukWat
EukWat1 = H2O_18S[,3:16] # remove OTU and Taxa columns
EukWat1[,3:14] = apply(EukWat1[,3:14], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
### Phyla ###
EukWat2 = data.frame(apply(EukWat1[,3:14], 2, function(x) aggregate (x~EukWat1$Phylum, FUN=sum))) # add phylum read count per each location
EukWat2 = EukWat2[,c(1,2,4,6,8,10,12,14,16,18,20,22,24)] # drop repeat phyla
colnames(EukWat2)[colnames(EukWat2)=="X10.EukWat1.Phylum"] = "Phylum"
EukWat3 = gather(EukWat2, "Phylum", 2:13) # aggregate by Phylum
names(EukWat3)=c("Phylum", "Location", "Richness") # relabel columns
EukWat_Phyla=aggregate(Richness~Phylum, data=EukWat3, FUN=sum) # Join loci
EukWat_Phyla # Look at table
# Top 10 Phyla (high to low) = Ochrophyta, Dinoflagellata, Florideophycidae, Unknown_Phylum_1002968, Ciliophora, Cnidaria, Protalveolata, Annelida, Basidiomycota, Cryptomonadales
# Rare (abundance 1) = Ascomycota, Brachiopoda, Nematoda, Platyhelminthes, Prymnesiophyceae, Echinodermata, Unknown_Phylum_1003810, SGUH942, Tunicata, SCM37C52
Ochrophyta=subset(EukWat3, Phylum == c("Ochrophyta"))
Dinoflagellata=subset(EukWat3, Phylum == c("Dinoflagellata"))
Florideophycidae=subset(EukWat3, Phylum == c("Florideophycidae"))
Unknown_Phylum_1002968=subset(EukWat3, Phylum == c("Unknown_Phylum_1002968"))
Ciliophora=subset(EukWat3, Phylum == c("Ciliophora"))
Cnidaria=subset(EukWat3, Phylum == c("Cnidaria"))
Protalveolata=subset(EukWat3, Phylum == c("Protalveolata"))
Annelida=subset(EukWat3, Phylum == c("Annelida"))
Basidiomycota=subset(EukWat3, Phylum == c("Basidiomycota"))
Cryptomonadales=subset(EukWat3, Phylum == c("Cryptomonadales"))
# Combine data sets
EukWat_PhylaOTU=rbind(Ochrophyta, Dinoflagellata, Florideophycidae, Unknown_Phylum_1002968, Ciliophora, Cnidaria, Protalveolata, Annelida, Basidiomycota, Cryptomonadales) 
# Create percentages so stacked bars add up to 100
EukWat_PhylaOTU2=aggregate(Richness~Location, EukWat_PhylaOTU, FUN=sum)
EukWat_PhylaOTU3=merge(EukWat_PhylaOTU, EukWat_PhylaOTU2, by = "Location", all = T)
EukWat_PhylaOTU3$Percentage=(EukWat_PhylaOTU3$Richness.x/EukWat_PhylaOTU3$Richness.y)*100
# OTU Phyla graph
EukWat_Phylagraph = 
  ggplot(data = EukWat_PhylaOTU3, aes(x = Location, y = Percentage, fill = Phylum)) +
  scale_fill_manual(values=c("#053061", "#92c5de", "#f7f7f7", "#e6f5d0", "#F65A00", "#b8e186", "#7fbc41", "#4d9221", "#F65A00", "#276419")) +
  geom_bar(stat = "identity")
ggsave("Figure3_EukWat_Phyla.png", dpi = 300)
### Classes ### 
EukWat4 = data.frame(apply(EukWat1[,3:14], 2, function(x) aggregate (x~EukWat1$Class, FUN=sum))) # add class read count per each location
EukWat4 = EukWat4[,c(1,2,4,6,8,10,12,14,16,18,20,22,24)] # drop repeat classes
colnames(EukWat4)[colnames(EukWat4)=="X10.EukWat1.Class"] = "Class"
EukWat5 = gather(EukWat4, "Class", 2:13) # aggregate by Class
names(EukWat5)=c("Class", "Location", "Richness") # relabel columns
EukWat_Class=aggregate(Richness~Class, data=EukWat5, FUN=sum) # Join loci
EukWat_Class # Look at table
# Top 10 Class (high to low) = Diatomea, Dinophyceae, Intramacronucleata, Rhodymeniophycidae, Chlorophyceae, Syndiniales, Polychaeta, Agaricomycetes, Anthozoa, Trebouxiophyceae
# Rare (abundance 1) = Dacrymycetes, Isochrysidales, Malacostraca, Prymnesiales, S9, Sordariomycetes, Unknown_Class_1003448
Diatomea=subset(EukWat5, Class == c("Diatomea"))
Dinophyceae=subset(EukWat5, Class == c("Dinophyceae"))
Intramacronucleata=subset(EukWat5, Class == c("Intramacronucleata"))
Rhodymeniophycidae=subset(EukWat5, Class == c("Rhodymeniophycidae"))
Chlorophyceae=subset(EukWat5, Class == c("Chlorophyceae"))
Syndiniales=subset(EukWat5, Class == c("Syndiniales"))
Polychaeta=subset(EukWat5, Class == c("Polychaeta"))
Agaricomycetes=subset(EukWat5, Class == c("Agaricomycetes"))
Anthozoa=subset(EukWat5, Class == c("Anthozoa"))
Trebouxiophyceae=subset(EukWat5, Class == c("Trebouxiophyceae"))
# Combine data sets
EukWat_ClassOTU=rbind(Diatomea, Dinophyceae, Intramacronucleata, Rhodymeniophycidae, Chlorophyceae, Syndiniales, Polychaeta, Agaricomycetes, Anthozoa, Trebouxiophyceae) 
# Create percentages so stacked bars add up to 100
EukWat_ClassOTU2=aggregate(Richness~Location, EukWat_ClassOTU, FUN=sum)
EukWat_ClassOTU3=merge(EukWat_ClassOTU, EukWat_ClassOTU2, by = "Location", all = T)
EukWat_ClassOTU3$Percentage=(EukWat_ClassOTU3$Richness.x/EukWat_ClassOTU3$Richness.y)*100
# OTU Class graph
EukWat_Classgraph = 
  ggplot(data = EukWat_ClassOTU3, aes(x = Location, y = Percentage, fill = Class)) +
  scale_fill_manual(values=c("#053061", "#2166ac", "#F65A00", "#f7f7f7", "#e6f5d0", "#b8e186", "#7fbc41", "#4d9221", "#F65A00", "darkred")) +
  geom_bar(stat = "identity")
ggsave("Figure3_EukWat_Class.png", dpi = 300)
########## EukWat ##########
######### Identify Abundant/Rare Phyla and Classes, Figure 3 ###########

# Adjust for final Figure 3 manually
```

Figure 4i: Principal Component Analyses (and permanova statistics)
```{r}
######### Packages ###########
# Upload the respective packages
install.packages("vegan")
library(vegan)
packageVersion("vegan") # v2.4.5
install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # v2.2.1
######### Packages ###########



######### Data and Clean-up ###########
# Caenogastropoda (cone snail) removed from all
# Contaminant Eukaryote in Prokaryote data removed from all
# Rarefied at 1000
install.packages("RCurl")
library("RCurl")
packageVersion("RCurl") # v1.95.4.10  *allows you to download from GitHub
# Prok = Prokaryotes (seawater & sediment microbes)
Prok=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_WatSed_1000.csv"))
# Euk = Eukaryotes (seawater & sediment microbes)
Euk=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_WatSed_1000.csv"))
# Metadata for Prok and Euk
Meta=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/Combined_Meta_1000.csv"))

# Merge data sets with metadata
Data_Prok=merge(Meta, Prok, by="Name")
Data_Euk=merge(Meta, Euk, by="ID")

# Subset by environment type
Sed_16S=subset(Data_Prok,Medium=="Sediment")
H2O_16S=subset(Data_Prok,Medium=="Water")
Sed_18S=subset(Data_Euk,Medium=="Sediment")
H2O_18S=subset(Data_Euk,Medium=="Water")
######### Data ###########



######### Statistics ###########
# Permanova statistics
adonis(Data_Prok[,5:2204]~Data_Prok$Medium*Data_Prok$Location, method="bray") # bray is a quantitative distance method 
# Medium: p = 0.001, Location p = 0.172, interaction: p = 0.219
# significant difference by medium
adonis(Data_Euk[,5:1242]~Data_Euk$Medium*Data_Euk$Location, method = "bray") 
# Medium: p = 0.001, Location: p = 0.040, interaction: p = 0.050
# significant difference by medium, location, and interaction  

# Pairwise function identifies which locations are different
pairwise.adonis <- function(x,factors, sim.method, p.adjust.m)
{
library(vegan)
co = as.matrix(combn(unique(factors),2))
pairs = c()
F.Model =c()
R2 = c()
p.value = c()
for(elem in 1:ncol(co)){
ad = adonis(x[factors %in% c(as.character(co[1,elem]),as.character(co[2,elem])),] ~
factors[factors %in% c(as.character(co[1,elem]),as.character(co[2,elem]))] , method =sim.method);
pairs = c(pairs,paste(co[1,elem],'vs',co[2,elem]));
F.Model =c(F.Model,ad$aov.tab[1,4]);
R2 = c(R2,ad$aov.tab[1,5]);
p.value = c(p.value,ad$aov.tab[1,6])
}
p.adjusted = p.adjust(p.value,method=p.adjust.m)
pairw.res = data.frame(pairs,F.Model,R2,p.value,p.adjusted)
return(pairw.res)
}

# Permanova statistics by environment type
adonis(Sed_16S[,5:2204]~Sed_16S$Location,method="bray") # p = 0.007
pairwise.adonis(Sed_16S[,5:2204], Sed_16S$Location, sim.method = "bray", p.adjust.m = "bonferroni") 
# Sediment: Major outlet proks is different than S1 p = 0.025
adonis(H2O_16S[,5:2204]~H2O_16S$Location,method="bray") # p = 0.016
pairwise.adonis(H2O_16S[,5:2204], H2O_16S$Location, sim.method = "bray", p.adjust.m = "bonferroni") 
# Seawater: Major outlet proks marginally different p=0.060-0.059
adonis(Sed_18S[,5:1242]~Sed_18S$Location,method="bray") # p = 0.099
pairwise.adonis(Sed_18S[,5:1242], Sed_18S$Location, sim.method = "bray", p.adjust.m = "bonferroni") 
# Sediment: No euks sites are different
adonis(H2O_18S[,5:1242]~H2O_18S$Location,method="bray") # p = 0.003
pairwise.adonis(H2O_18S[,5:1242], H2O_18S$Location, sim.method = "bray", p.adjust.m = "bonferroni") 
# Seawater: All euks different 
######### Statistics ###########



######### PCAs ###########
Sed_16S=Sed_16S[,5:2204]
H2O_16S=H2O_16S[,5:2204]
Sed_18S=Sed_18S[,5:1242]
H2O_18S=H2O_18S[,5:1242]

# Create PCAs
ProkSed.PCA<-prcomp(Sed_16S)
summary(ProkSed.PCA) # Comps 1(.25), 2(.14) = 39% variance
ProkWat.PCA<-prcomp(H2O_16S)
summary(ProkWat.PCA) # Comps 1(.46), 2(.27) = 73% variance
EukSed.PCA<-prcomp(Sed_18S)
summary(EukSed.PCA) # Comps 1(.42), 2(.16) = 58% variance
EukWat.PCA<-prcomp(H2O_18S)
summary(EukWat.PCA) # Comps 1(.62), 2(.17) = 79% variance

# Adjust stat_ellipse function so n=3 or more for ellipse (n=4 before)
stat_ellipse <- function(mapping = NULL, data = NULL,
                         geom = "path", position = "identity",
                         ...,
                         type = "t",
                         level = 0.95,
                         segments = 51,
                         na.rm = FALSE,
                         show.legend = NA,
                         inherit.aes = TRUE) {
  layer(
    data = data,
    mapping = mapping,
    stat = StatEllipse,
    geom = geom,
    position = position,
    show.legend = show.legend,
    inherit.aes = inherit.aes,
    params = list(
      type = type,
      level = level,
      segments = segments,
      na.rm = na.rm,
      ...
    )
  )
}

StatEllipse <- ggproto("StatEllipse", Stat,
  required_aes = c("x", "y"),

  compute_group = function(data, scales, type = "t", level = 0.95,
                           segments = 51, na.rm = FALSE) {
    calculate_ellipse(data = data, vars = c("x", "y"), type = type,
                      level = level, segments = segments)
  }
)

calculate_ellipse <- function(data, vars, type, level, segments){
  dfn <- 2
  dfd <- nrow(data) - 1

  if (!type %in% c("t", "norm", "euclid")) {
    message("Unrecognized ellipse type")
    ellipse <- rbind(as.numeric(c(NA, NA)))
  } else if (dfd < 2) { #dfd changed from 3 to 2
    message("Too few points to calculate an ellipse")
    ellipse <- rbind(as.numeric(c(NA, NA)))
  } else {
    if (type == "t") {
      v <- MASS::cov.trob(data[,vars])
    } else if (type == "norm") {
      v <- stats::cov.wt(data[,vars])
    } else if (type == "euclid") {
      v <- stats::cov.wt(data[,vars])
      v$cov <- diag(rep(min(diag(v$cov)), 2))
    }
    shape <- v$cov
    center <- v$center
    chol_decomp <- chol(shape)
    if (type == "euclid") {
      radius <- level/max(chol_decomp)
    } else {
      radius <- sqrt(dfn * stats::qf(level, dfn, dfd))
    }
    angles <- (0:segments) * 2 * pi/segments
    unit.circle <- cbind(cos(angles), sin(angles))
    ellipse <- t(center + radius * t(unit.circle %*% chol_decomp))
  }

  ellipse <- as.data.frame(ellipse)
  colnames(ellipse) <- vars
  ellipse
}
######### PCAs ###########



######### PCA Plots for Figure 4i ###########
# ProkSed PCA
Sed_Meta=Meta[13:23,]
ProkSed_Meta=data.frame(ProkSed.PCA$x, 
                   Type=Sed_Meta$Medium, 
                   Location=Sed_Meta$Location)
ProkSed=
  ggplot(data=ProkSed_Meta, 
         x=PC1, 
         y=PC2, 
         col=Type) +
  geom_point(data=ProkSed_Meta, 
             aes(x=PC1, 
                 y=PC2, 
                 col=Location),
             size=5) +
   stat_ellipse(geom = "polygon",
                alpha = 0.35,
                aes(x=PC1, # default normal confidence is 95% 
                    y=PC2, 
                    fill=Location,
                    color=Location)) +
  scale_fill_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  scale_color_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  theme(plot.title = element_text(size = 40))
# http://ggplot2.tidyverse.org/reference/ggtheme.html
ProkSed=ProkSed+ggtitle("Bacteria and Archaea (Sediment)")+ theme_set(theme_linedraw())
ggsave("Figure4i_ProkSed.png", dpi = 300)
# ProkWat PCA
Wat_Meta=Meta[1:12,]
ProkWat_Meta=data.frame(ProkWat.PCA$x, 
                   Type=Wat_Meta$Medium, 
                   Location=Wat_Meta$Location)
ProkWat=
  ggplot(data=ProkWat_Meta, 
         x=PC1, 
         y=PC2, 
         col=Type) +
  geom_point(data=ProkWat_Meta, 
             aes(x=PC1, 
                 y=PC2, 
                 col=Location),
             size=5) +
   stat_ellipse(geom = "polygon",
                alpha = 0.15,
                aes(x=PC1, # default normal confidence is 95% 
                    y=PC2, 
                    fill=Location)) +
  scale_fill_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  scale_color_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  theme_set(theme_linedraw()) # http://ggplot2.tidyverse.org/reference/ggtheme.html
ProkWat=ProkWat+ggtitle("Bacteria and Archaea (Seawater)")+ theme_set(theme_linedraw())
ggsave("Figure4i_ProkWat.png", dpi = 300)
# EukSed PCA
Sed_Meta=Meta[14:22,]
EukSed_Meta=data.frame(EukSed.PCA$x, 
                   Type=Sed_Meta$Medium, 
                   Location=Sed_Meta$Location)
EukSed=
  ggplot(data=EukSed_Meta, 
         x=PC1, 
         y=PC2, 
         col=Type) +
  geom_point(data=EukSed_Meta, 
             aes(x=PC1, 
                 y=PC2, 
                 col=Location),
             size=5) +
   stat_ellipse(geom = "polygon",
                alpha = 0.35,
                aes(x=PC1, # default normal confidence is 95% 
                    y=PC2, 
                    fill=Location,
                    color=Location)) +
  scale_fill_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  scale_color_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  theme_set(theme_linedraw()) # http://ggplot2.tidyverse.org/reference/ggtheme.html
EukSed=EukSed+ggtitle("Eukaryotes (Sediment)")+ theme_set(theme_linedraw())
ggsave("Figure4i_EukSed.png", dpi = 300)
# EukWat PCA
Wat_Meta=Meta[1:12,]
EukWat_Meta=data.frame(EukWat.PCA$x, 
                   Type=Wat_Meta$Medium, 
                   Location=Wat_Meta$Location)
EukWat=
  ggplot(data=EukWat_Meta, 
         x=PC1, 
         y=PC2, 
         col=Type) +
  geom_point(data=EukWat_Meta, 
             aes(x=PC1, 
                 y=PC2, 
                 col=Location),
             size=5) +   stat_ellipse(geom = "polygon",
                alpha = 0.15,
                aes(x=PC1, # default normal confidence is 95% 
                    y=PC2, 
                    fill=Location)) +
  scale_fill_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  scale_color_manual(values=c("#F65A00", "#68AF2B", "#0202DE")) +
  theme_set(theme_linedraw()) # http://ggplot2.tidyverse.org/reference/ggtheme.html
EukWat=EukWat+ggtitle("Eukaryotes (Seawater)")+ theme_set(theme_linedraw())
ggsave("Figure4i_EukWat.png", dpi = 300)
######### PCA Plots for Figure 4i ###########

# Adjust for final Figure 4i manually
```

Figure 4ii: Venn diagrams
```{r}
######### Packages ###########
# Upload the respective packages
install.packages("VennDiagram")
library(VennDiagram)
packageVersion("VennDiagram") # v1.6.20
######### Packages ###########



######### Data and Clean-up ###########
# Caenogastropoda (cone snail) removed from all
# Contaminant Eukarytote in Prokaryote data removed from all
# Rarefied at 1000
install.packages("RCurl")
library("RCurl")
packageVersion("RCurl") # v1.95.4.10  *allows you to download from GitHub
# Prok = Prokaryotes (seawater & sediment microbes)
Prok=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_WatSed_1000.csv"))
# Euk = Eukaryotes (seawater & sediment microbes)
Euk=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_WatSed_1000.csv"))
# Metadata for Prok and Euk
Meta=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/Combined_Meta_1000.csv"))

# Merge data sets with metadata
Data_Prok=merge(Meta,Prok,by="Name")
Data_Euk=merge(Meta,Euk,by="ID")
######### Data ###########



######### Aggregate by Medium and Location for Venn Diagrams ###########
# Proks
venn.df.Prok=aggregate(Data_Prok[,5:2204], by = list(Data_Prok$Medium, Data_Prok$Location), FUN = "sum")

venn.df.Prok[,1:7] # check row order, sediment should be rows 1,3,5
venn.df.Prok=venn.df.Prok[,-1] # Get rid of "Group.1"
venn.df.Prok=venn.df.Prok[,-1] # Get rid of "Group.2"

# Euks
venn.df.Euk=aggregate(Data_Euk[,5:1242], by = list(Data_Euk$Medium, Data_Euk$Location), FUN = "sum")

venn.df.Euk[,1:7] # check row order, sediment should be rows 1,3,5
venn.df.Euk=venn.df.Euk[,-1] # Get rid of "Group.1"
venn.df.Euk=venn.df.Euk[,-1] # Get rid of "Group.2"
######### Aggregate by Medium and Location for Venn Diagrams ###########



######### Venn diagrams for Figure 4ii ###########
# ProkSed
ProkSed.MjO = colnames(venn.df.Prok[5,apply(venn.df.Prok[5,], MARGIN=2, function(x) any(x >0))])
ProkSed.MnO = colnames(venn.df.Prok[1, apply(venn.df.Prok[1,], MARGIN=2, function(x) any(x >0))])
ProkSed.Sh = colnames(venn.df.Prok[3, apply(venn.df.Prok[3,], MARGIN=2, function(x) any(x >0))])

venn.diagram(x = list(ProkSed.MjO, ProkSed.MnO, ProkSed.Sh), filename = "Figure4ii_ProkSed.png", category = c("Major Outlet", "Minor Outlet", "Sheltered"), print.mode = "percent", sigdigs = 2 , cex= 1.5, fill = c("#F65A00", "#797171", "#000000"), alpha = 0.6)
# ProkWat
ProkWat.MjO = colnames(venn.df.Prok[6,apply(venn.df.Prok[6,], MARGIN=2, function(x) any(x >0))])
ProkWat.MnO = colnames(venn.df.Prok[2, apply(venn.df.Prok[2,], MARGIN=2, function(x) any(x >0))])
ProkWat.Sh = colnames(venn.df.Prok[4, apply(venn.df.Prok[4,], MARGIN=2, function(x) any(x >0))])

venn.diagram(x = list(ProkWat.MjO, ProkWat.MnO, ProkWat.Sh), filename = "Figure4ii_ProkWat.png", category = c("Major Outlet", "Minor Outlet", "Sheltered"), print.mode = "percent", sigdigs = 2 , cex= 1.5, fill = c("#F65A00", "#797171", "#0202DE"), alpha = 0.3)
# EukSed
EukSed.MjO = colnames(venn.df.Euk[5,apply(venn.df.Euk[5,], MARGIN=2, function(x) any(x >0))])
EukSed.MnO = colnames(venn.df.Euk[1, apply(venn.df.Euk[1,], MARGIN=2, function(x) any(x >0))])
EukSed.Sh = colnames(venn.df.Euk[3, apply(venn.df.Euk[3,], MARGIN=2, function(x) any(x >0))])

venn.diagram(x = list(EukSed.MjO, EukSed.MnO, EukSed.Sh), filename = "Figure4ii_EukSed.png", category = c("Major Outlet", "Minor Outlet", "Sheltered"), print.mode = "percent", sigdigs = 2, cex= 1.5, fill = c("#F65A00", "#68AF2B", "#0202DE"), alpha = 0.6) 
# EukWat
EukWat.MjO = colnames(venn.df.Euk[6,apply(venn.df.Euk[6,], MARGIN=2, function(x) any(x >0))])
EukWat.MnO = colnames(venn.df.Euk[2, apply(venn.df.Euk[2,], MARGIN=2, function(x) any(x >0))])
EukWat.Sh = colnames(venn.df.Euk[4, apply(venn.df.Euk[4,], MARGIN=2, function(x) any(x >0))])

venn.diagram(x = list(EukWat.MjO, EukWat.MnO, EukWat.Sh), filename = "Figure4ii_EukWat.png", category = c("Major Outlet", "Minor Outlet", "Sheltered"), print.mode = "percent", sigdigs = 2 , cex= 1.5, fill = c("#FFFFFF", "#797171", "#000000"), alpha = 0.3) 
######### Venn diagrams for Figure 4ii ###########

# Adjust for final Figure 4ii manually
```

Figure 5: Heatmaps of driving taxa (all classes from abundant phyla)
```{r}
######### Packages ###########
install.packages("dplyr")
library(dplyr)
packageVersion("dplyr") # v0.7.4
install.packages("reshape2")
library(reshape2)
packageVersion("reshape2") # v1.4.3
install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # v2.2.1
install.packages("tidyr")
library("tidyr")
packageVersion("tidyr") # v0.7.2
######### Packages ###########



######### Data and Clean-up ###########
# Caenogastropoda (cone snail) removed from all
# Contaminant Eukaryote in Prokaryote data removed from all
# Rarefied at 1000
install.packages("RCurl")
library("RCurl")
packageVersion("RCurl") # v1.95.4.10  *allows you to download from GitHub
# Prok = Prokaryotes (seawater & sediment microbes)
Prok=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_WatSed_1000.csv"))
# Euk = Eukaryotes (seawater & sediment microbes)
Euk=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_WatSed_1000.csv"))
# Metadata for Prok and Euk
Meta=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/Combined_Meta_1000.csv"))

Prok[Prok>1]=1 # Set abundance to 1 "‘>’ not meaningful for factors" still works
Euk[Euk>1]=1 # Set abundance to 1 "‘>’ not meaningful for factors" still works

# Merge data sets with metadata
Data_Prok=merge(Meta,Prok,by="Name")
Data_Euk=merge(Meta,Euk,by="ID")

# Subset by environment type
Sed_16S=subset(Data_Prok,Medium=="Sediment")
H2O_16S=subset(Data_Prok,Medium=="Water")
Sed_18S=subset(Data_Euk,Medium=="Sediment")
H2O_18S=subset(Data_Euk,Medium=="Water")

# Upload OTU taxa tables
# Prok Phyla and Class
ProkTaxa=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_Taxa.csv"))
# Euk Phyla and Class
EukTaxa=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_Taxa.csv"))

# Add X so all OTU match
ProkTaxa$OTU <- sub("^", "X", ProkTaxa$OTU )
EukTaxa$OTU <- sub("^", "X", EukTaxa$OTU )

# Transpose all environment type data sets
Sed_16S=as.data.frame(t(Sed_16S))
H2O_16S=as.data.frame(t(H2O_16S))
Sed_18S=as.data.frame(t(Sed_18S))
H2O_18S=as.data.frame(t(H2O_18S))

# Create a new column out of left and label as "OTU" 
Sed_16S = add_rownames(Sed_16S, "OTU")
H2O_16S = add_rownames(H2O_16S, "OTU")
Sed_18S = add_rownames(Sed_18S, "OTU")
H2O_18S = add_rownames(H2O_18S, "OTU")

Sed_16S=merge(ProkTaxa,Sed_16S,by="OTU") # 13-16 SH, 17-19 MN, 20-23 MJ
H2O_16S=merge(ProkTaxa,H2O_16S,by="OTU") # 1-4 SH, 5-8 MN, 9-12 MJ
Sed_18S=merge(EukTaxa,Sed_18S,by="OTU") # 1-3 SH, 4-6 MN, 7-9 MJ
H2O_18S=merge(EukTaxa,H2O_18S,by="OTU") # 10-13 SH, 14-17 MN, 18-21 MJ
######### Data and Clean-up ###########



######### Filter data for only abundant phyla ###########
##### ProkSed ##### 
ProkSed1 = Sed_16S[,3:15] # remove OTU and Taxa columns
ProkSed1[,3:13] = apply(ProkSed1[,3:13], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
# Top 10 Phyla
Bacteroidetes=subset(ProkSed1, Phylum == c("Bacteroidetes"))
Proteobacteria=subset(ProkSed1, Phylum == c("Proteobacteria"))
Planctomycetes=subset(ProkSed1, Phylum == c("Planctomycetes"))
Acidobacteria=subset(ProkSed1, Phylum == c("Acidobacteria"))
Actinobacteria=subset(ProkSed1, Phylum == c("Actinobacteria"))
Chloroflexi=subset(ProkSed1, Phylum == c("Chloroflexi"))
Verrucomicrobia=subset(ProkSed1, Phylum == c("Verrucomicrobia"))
Lentisphaerae=subset(ProkSed1, Phylum == c("Lentisphaerae"))
Firmicutes=subset(ProkSed1, Phylum == c("Firmicutes"))
Chlorobi=subset(ProkSed1, Phylum == c("Chlorobi"))
# Combine data sets
ProkSed_Abd=rbind(Bacteroidetes, Proteobacteria, Planctomycetes, Acidobacteria, Actinobacteria, Chloroflexi, Verrucomicrobia, Lentisphaerae, Firmicutes, Chlorobi)
# Adjust data set
ProkSed2 = data.frame(apply(ProkSed_Abd[,3:13], 2, function(x) aggregate (x~ProkSed_Abd$Class, FUN=sum))) # add class read count per each location
ProkSed2 = ProkSed2[,c(1,2,4,6,8,10,12,14,16,18,20,22)] # drop the repeat phyla
ProkSed3 = gather(ProkSed2, "Location", "Richness", 2:12)
names(ProkSed3)=c("Class", "Location", "Richness") 
ProkSed_keep = subset(ProkSed3, ! Richness == 0 ) # keep values > 0
##### ProkSed ##### 
##### ProkWat ##### 
ProkWat1 = H2O_16S[,3:16] # remove OTU and Taxa columns
ProkWat1[,3:14] = apply(ProkWat1[,3:14], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
# Top 10 Phyla
Proteobacteria=subset(ProkWat1, Phylum == c("Proteobacteria"))
Bacteroidetes=subset(ProkWat1, Phylum == c("Bacteroidetes"))
Planctomycetes=subset(ProkWat1, Phylum == c("Planctomycetes"))
Firmicutes=subset(ProkWat1, Phylum == c("Firmicutes"))
Verrucomicrobia=subset(ProkWat1, Phylum == c("Verrucomicrobia"))
Cyanobacteria=subset(ProkWat1, Phylum == c("Cyanobacteria"))
Lentisphaerae=subset(ProkWat1, Phylum == c("Lentisphaerae"))
Actinobacteria=subset(ProkWat1, Phylum == c("Actinobacteria"))
Spirochaetae=subset(ProkWat1, Phylum == c("Spirochaetae"))
Woesearchaeota_DHVEG6=subset(ProkWat1, Phylum == c("Woesearchaeota__DHVEG_6_"))
# Combine data sets
ProkWat_Abd=rbind(Proteobacteria, Bacteroidetes, Planctomycetes, Firmicutes, Verrucomicrobia, Cyanobacteria, Lentisphaerae, Actinobacteria, Spirochaetae, Woesearchaeota_DHVEG6)
# Adjust data set
ProkWat2 = data.frame(apply(ProkWat_Abd[,3:14], 2, function(x) aggregate (x~ProkWat_Abd$Class, FUN=sum))) # add class read count per each location
ProkWat2 = ProkWat2[,c(1,2,4,6,8,10,12,14,16,18,20,22,24)] # drop the repeat phyla
ProkWat3 = gather(ProkWat2, "Location", "Richness", 2:13)
names(ProkWat3)=c("Class", "Location", "Richness") 
ProkWat_keep = subset(ProkWat3, ! Richness == 0 ) # keep values > 0
##### ProkWat ##### 
##### EukSed ##### 
EukSed1 = Sed_18S[,3:13] # remove OTU and Taxa columns
EukSed1[,3:11] = apply(EukSed1[,3:11], 2, function(x) as.numeric(as.character(x))) # convert characters to numeric
# Top 10 Phyla
Ochrophyta=subset(EukSed1, Phylum == c("Ochrophyta"))
Dinoflagellata=subset(EukSed1, Phylum == c("Dinoflagellata"))
Ciliophora=subset(EukSed1, Phylum == c("Ciliophora"))
Basidiomycota=subset(EukSed1, Phylum == c("Basidiomycota"))
Cnidaria=subset(EukSed1, Phylum == c("Cnidaria"))
Florideophycidae=subset(EukSed1, Phylum == c("Florideophycidae"))
Annelida=subset(EukSed1, Phylum == c("Annelida"))
Arthropoda=subset(EukSed1, Phylum == c("Arthropoda"))
Apicomplexa=subset(EukSed1, Phylum == c("Apicomplexa"))
Unknown_Phylum_1002968=subset(EukSed1, Phylum == c("Unknown_Phylum_1002968"))
# Combine data sets
EukSed_Abd=rbind(Ochrophyta, Dinoflagellata, Ciliophora, Basidiomycota, Cnidaria, Florideophycidae, Annelida, Arthropoda, Apicomplexa, Unknown_Phylum_1002968)
# Adjust data set
EukSed2 = data.frame(apply(EukSed_Abd[,3:11], 2, function(x) aggregate (x~EukSed_Abd$Class, FUN=sum))) # add class read count per each location
EukSed2 = EukSed2[,c(1,2,4,6,8,10,12,14,16,18)] # drop the repeat phyla
EukSed3 = gather(EukSed2, "Location", "Richness", 2:10)
names(EukSed3)=c("Class", "Location", "Richness") 
EukSed_keep = subset(EukSed3, ! Richness == 0 ) # keep values > 0
##### EukSed ##### 
##### EukWat ##### 
EukWat1 = H2O_18S[,3:16] # remove OTU and Taxa columns
EukWat1[,3:14] = apply(EukWat1[,3:14], 2, function(x) as.numeric(as.character(x)))
# Top 10 Phyla
Ochrophyta=subset(EukWat1, Phylum == c("Ochrophyta"))
Dinoflagellata=subset(EukWat1, Phylum == c("Dinoflagellata"))
Florideophycidae=subset(EukWat1, Phylum == c("Florideophycidae"))
Unknown_Phylum_1002968=subset(EukWat1, Phylum == c("Unknown_Phylum_1002968"))
Ciliophora=subset(EukWat1, Phylum == c("Ciliophora"))
Cnidaria=subset(EukWat1, Phylum == c("Cnidaria"))
Protalveolata=subset(EukWat1, Phylum == c("Protalveolata"))
Annelida=subset(EukWat1, Phylum == c("Annelida"))
Basidiomycota=subset(EukWat1, Phylum == c("Basidiomycota"))
Cryptomonadales=subset(EukWat1, Phylum == c("Cryptomonadales"))
# Combine data sets
EukWat_Abd=rbind(Ochrophyta, Dinoflagellata, Florideophycidae, Unknown_Phylum_1002968, Ciliophora, Cnidaria, Protalveolata, Annelida, Basidiomycota, Cryptomonadales)
# Adjust data set
EukWat2 = data.frame(apply(EukWat_Abd[,3:14], 2, function(x) aggregate (x~EukWat_Abd$Class, FUN=sum))) # add class read count per each location
EukWat2 = EukWat2[,c(1,2,4,6,8,10,12,14,16,18,20,22,24)] # drop the repeat phyla
EukWat3 = gather(EukWat2, "Location", "Richness", 2:13)
names(EukWat3)=c("Class", "Location", "Richness") 
EukWat_keep = subset(EukWat3, ! Richness == 0 ) # keep values > 0
##### EukWat ##### 
######### Filter data for only abundant phyla ###########



######### Heatmaps for Figure 5 ###########
# ProkSed
ProkSed_heat = ggplot(ProkSed_keep, aes(Location, Class)) +
  geom_tile(aes(fill = Richness), color = "black") +
  scale_fill_gradient(low = "lightblue", high = "red") +
  ylab("List of Class") +
  xlab("List of Location") +
  theme(legend.title = element_text(size = 10),
        legend.text = element_text(size = 12),
        plot.title = element_text(size=16),
        axis.title=element_text(size=14,face="bold"),
        axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(fill = "Richness")
ggsave("Figure5_ProkSed_Abd.png", dpi = 300)
# ProkWat
ProkWat_heat = ggplot(ProkWat_keep, aes(Location, Class)) +
  geom_tile(aes(fill = Richness), color = "black") +
  scale_fill_gradient(low = "lightblue", high = "red") +
  ylab("List of Class") +
  xlab("List of Location") +
  theme(legend.title = element_text(size = 10),
        legend.text = element_text(size = 12),
        plot.title = element_text(size=16),
        axis.title=element_text(size=14,face="bold"),
        axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(fill = "Richness")
ggsave("Figure5_ProkWat_Abd.png", dpi = 300)
# EukSed
EukSed_heat = ggplot(EukSed_keep, aes(Location, Class)) +
  geom_tile(aes(fill = Richness), color = "black") +
  scale_fill_gradient(low = "lightblue", high = "red") +
  ylab("List of Class") +
  xlab("List of Location") +
  theme(legend.title = element_text(size = 10),
        legend.text = element_text(size = 12),
        plot.title = element_text(size=16),
        axis.title=element_text(size=14,face="bold"),
        axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(fill = "Richness")
ggsave("Figure5_EukSed_Abd.png", dpi = 300)
# EukWat
EukWat_heat = ggplot(EukWat_keep, aes(Location, Class)) +
  geom_tile(aes(fill = Richness), color = "black") +
  scale_fill_gradient(low = "lightblue", high = "red") +
  ylab("List of Class") +
  xlab("List of Location") +
  theme(legend.title = element_text(size = 10),
        legend.text = element_text(size = 12),
        plot.title = element_text(size=16),
        axis.title=element_text(size=14,face="bold"),
        axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(fill = "Richness")
ggsave("Figure5_EukWat_Abd.png", dpi = 300)
######### Heatmaps for Figure 5 ###########

# Adjust for final Figure 4ii manually
```

Supplemental: Diversity Indices and Rarefaction Curves 
```{r}
######### Packages ###########
# Upload the respective packages
install.packages("vegan")
library("vegan")
packageVersion("vegan")     # v2.4.5
install.packages("ggplot2")
library("ggplot2")
packageVersion("ggplot2") 
######### Packages ###########



######### Data and Clean-up ###########
# Caenogastropoda (cone snail) removed from all
# Contaminant Eukaryote in Prokaryote data removed from all
# Rarefied at 1000
# Prok = Prokaryotes (seawater & sediment microbes)
Prok=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/16S_WatSed_1000.csv"))
ProkData=(Prok[,2:2201])
# Euk = Eukaryotes (seawater & sediment microbes)
Euk=read.csv(text=getURL("https://raw.githubusercontent.com/sul-hasan/MicrobialCommunityAnalysesPipeline_Supplemental_SUHetal/master/18S_WatSed_1000.csv"))
EukData=(Euk[,2:1239])
######### Data and Clean-up ###########



######### Statistics ###########
# Carrying the statistical analyses
# Shannon Index (abundance)
Prok_Shan=diversity(ProkData)
Prok_Shan
Euk_Shan=diversity(EukData)
Euk_Shan
# Simpson Index (abundance)
Prok_Simp=diversity(ProkData, "simpson")
Prok_Simp
Euk_Simp=diversity(EukData, "simpson")
Euk_Simp
######### Statistics ###########



######### Plot diversity indices ###########
png("Supp_ProkShannon.png.png")
hist(Prok_Shan)
dev.off()
png("Supp_ProkSimpson.png")
hist(Prok_Simp)
dev.off()
png("Supp_EukShannon.png.png")
hist(Euk_Shan)
dev.off()
png("Supp_EukSimpson.png")
hist(Euk_Simp)
dev.off()
######### Plot diversity indices ###########



######### Plot rarefaction curves ###########
# Set-up plotting parameters
col=c("black", "darkred", "forestgreen", "orange", "blue", "yellow", "hotpink")
lty=c("solid", "dashed", "longdash", "dotdash")
pars=expand.grid(col = col, lty = lty, stringsAsFactors = FALSE)
head(pars)

P_max=min(rowSums(ProkData)) # number of individual in each plot 
P_max                        # use smallest number of observations
png("Supp_Prok_RareCurve.png")
Prok_RareCurve=with(pars[1:28, ],
            rarecurve(ProkData, step = 20, sample = P_max, col = col,
                      xlab = "Number of 16S Reads", ylab = "Number of OTUs"))
dev.off()

E_max=min(rowSums(EukData)) 
E_max
png("Supp_Euk_RareCurve.png")
Euk_RareCurve=with(pars[1:28, ],
            rarecurve(EukData, step = 20, sample = E_max, col = col,
                      xlab = "Number of 18S Reads", ylab = "Number of OTUs"))
dev.off()
######### Plot rarefaction curves ###########
```